sdk/src/main/java/com/google/cloud/dataflow/sdk/io/OffsetBasedSource.java

```diff
@@ -264,7 +264,18 @@ public final boolean start() throws IOException {
 
     @Override
     public final boolean advance() throws IOException {
-      return advanceImpl() && rangeTracker.tryReturnRecordAt(isAtSplitPoint(), getCurrentOffset());
+      if (!advanceImpl()) {
+        // Reader could not read another record. Add an extra split point in order to "finish" the
```

Comment:
I thought about that approach, but I decided that this one makes most sense. For example, we'll get an error if the user calls this twice with the same value (vs adding a boolean to track EOF called twice).